### PR TITLE
Refactor and improve code Quality

### PR DESCRIPTION
**Pull request by TheStachelfisch**
These if statements with hundreds of || are disgusting. Use Linq

And use new C# features

Thanks

### CODE REVIEW

File: ModInstances/SpookyNPC.cs [hunk 1]
```diff
@@ -15,128 +15,93 @@
...
+            if (ModContent.GetInstance<SpookyTerraria>().beatGame ||
+                Main.worldName == SpookyTerrariaUtils.slenderWorldName)
                 pool.Clear(); // No more enemy spawns!.. Because you beat the game, numnut
-            }
-            if (!ModContent.GetInstance<SpookyConfigServer>().excludeAllNPCsBesideCreepyTerrariaOnesFromSpawnPool)
-            {
-                return;
-            }
+
+            if (!ModContent.GetInstance<SpookyConfigServer>().excludeAllNPCsBesideCreepyTerrariaOnesFromSpawnPool) return;
```
Comment (RighteousRyan1) [inline ModInstances/SpookyNPC.cs:54]: These oneliners you changed my code to are quite disgusting

Comment (TheStachelfisch) [reply ModInstances/SpookyNPC.cs:54]: I mean, you can change the syntax style back... But the || if statements are extremely disgusting and have to be replaced with something more efficient

File: ModInstances/SpookyNPC.cs [hunk 2]
```diff
@@ -315,57 +279,57 @@ public override void GetChat(NPC npc, ref string chat)
...
-            }
         }
+
         public static int slenderFailsafeTimer;
+
         public override void PostAI(NPC npc)
         {
-            Player player = Main.player[Main.myPlayer];
-            float distanceToStalker = npc.Distance(player.Center);
+            var player = Main.player[Main.myPlayer];
```
Comment (RighteousRyan1) [inline ModInstances/SpookyNPC.cs:288]: also not really enjoying all of these swaps from the actual type -> `var`

**Comment by RighteousRyan1** [comment]
The if statements were completely gross, yes. My lack of knowledge with
Linq caused that
***@***.***>:

File: ModInstances/SpookyNPC.cs (hunk 2 above)
Comment (TheStachelfisch) [reply ModInstances/SpookyNPC.cs:288]: var's are more efficient in this context

File: ModInstances/SpookyWorld.cs [hunk 3]
```diff
@@ -112,105 +106,90 @@ public override void PostDrawTiles()
...
-                    if (leftTile.type == TileID.Trees && !leftMostTile.active() && leftTile.frameX == 0 && leftTile.frameY < 66)
-					{
-                        leftSideSpots.Add(new Point(i, j));
-					}
-                    else if (rightTile.type == TileID.Trees && !rightMostTile.active() && rightTile.frameX == 0 && rightTile.frameY < 66)
-					{
-                        rightSideSpots.Add(new Point(i, j));
-					}
-                }
+
```
Comment (RighteousRyan1) [inline ModInstances/SpookyWorld.cs:147]: Idfk what this was, github fucking broke the indentations here, they don’t exist on my build

Comment (TheStachelfisch) [reply ModInstances/SpookyWorld.cs:147]: Hmm yeah. Github likes to do that

**Pull request merged**